arch/arm/core/cortex_m/mpu/arm_mpu_v8_internal.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2017 Linaro Limited.
+ * Copyright (c) 2018 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* This internal function performs MPU region initialization.
+ *
+ * Note:
+ *   The caller must provide a valid region index.
+ */
+static void _region_init(u32_t index, struct arm_mpu_region *region_conf)
+{
+	/* Select the region you want to access */
+	MPU->RNR = index;
+	/* Configure the region */
+	MPU->MAIR0 = region_conf->attr.mair & MPU_MAIR0_Attr0_Msk;
+	MPU->RBAR = (region_conf->base & MPU_RBAR_BASE_Msk)
+		| (region_conf->attr.rbar &
+			(MPU_RBAR_XN_Msk | MPU_RBAR_AP_Msk | MPU_RBAR_SH_Msk));
+	MPU->RLAR = (region_conf->attr.rlar & MPU_RLAR_LIMIT_Msk)
+		| MPU_RLAR_EN_Msk;
+	SYS_LOG_DBG("[%d] 0x%08x 0x%08x 0x%08x 0x%08x",
+			index, region_conf->base, region_conf->attr.rbar,
+			region_conf->attr.mair, region_conf->attr.rlar);
+}
+
+#if defined(CONFIG_USERSPACE) || defined(CONFIG_MPU_STACK_GUARD) || \
+	defined(CONFIG_APPLICATION_MEMORY)
+
+/**
+ * This internal function allocates default RAM cache-ability, share-ability,
+ * and execution allowance attributes along with the requested access
+ * permissions and size.
+ */
+static inline arm_mpu_region_attr_t _get_mpu_ram_region_attr(u32_t ap,
+	u32_t base, u32_t size)
+{
+	arm_mpu_region_attr_t attr;
+
+	attr.rbar = ((1 << MPU_RBAR_XN_Pos) & MPU_RBAR_XN_Msk)
```

Comment:
Well, here, I think it can stay, because, in the definition of the struct we have:
`struct arm_mpu_region_attr {
	/* Attributes belonging to RBAR */
	u8_t rbar: 5;`

So the documentation of the field indicates clearly that this is just the attributes that go to RBAR; not the base address.

And we always use this as: `attr.rbar`, in the source code.